### PR TITLE
Upgrade Reactabular to 8.17.0

### PR DESCRIPTION
**Pull request by islemaster**
Reverts code-dot-org/code-dot-org#30537, restoring https://github.com/code-dot-org/code-dot-org/pull/30493.

The last attempt broke the ability to edit student names in the ManageStudentsTable component.
image

Root cause: This component references props directly in its formatter that weren't passed through the table component, here:
https://github.com/code-dot-org/code-dot-org/blob/24c9bd6dcba3a0150e1ec67062ed1454f4f89440/apps/src/templates/manageStudents/ManageStudentsTable.jsx#L261
Ideally formatters should be pure functions, and any changing state should be included in the rows passed to the Reactabular `Body` component so it can update appropriately.

[This article](https://medium.com/@fabianterh/investigating-enzymes-keypress-simulation-on-input-fields-d0cbf0f2ca3f) was helpful for writing a test that actually checks whether a field within the table was editable.

I've done a review of all my changes in this PR again to look for other places where formatters may be incorrectly depending on external state.  I only found one other problem case: The `CensusInaccuracyReviewTable`.  I'm not even sure it was a problem in practice, but the test I wrote shows the 'broken' behavior.

### CODE REVIEW

File: apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js [hunk 1]
```diff
@@ -142,5 +144,40 @@ describe('ManageStudentsTable', () => {
...
+          .first();
+      expect(nameInput().prop('value')).to.equal(fakeStudent.name);
+
+      // Simulate a name change
+      nameInput().simulate('change', {target: {value: fakeStudent.name + 'z'}});
+
+      // Expect the input box value to have changed
+      expect(nameInput().prop('value')).to.equal(fakeStudent.name + 'z');
+    });
   });
```
Comment (Erin007) [inline apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js:182]: Thanks for the increased test coverage here!

**Comment by Erin007** [comment]
Is the `CensusInaccuracyReviewTable` still in use?

**Review by Erin007 — APPROVED** [review]

**Comment by islemaster** [comment]
We still have a view that loads it; I think it's seasonal, and may be useful again here in a few months.

**Review by jmkulwik — COMMENTED** [review]
Looks good. Would it make more sense to make this migration in chunks rather than all at once given how many regressions it has hit the various times we've tried to do it? Or are you pretty confident this time?

**Comment by islemaster** [comment]
I've got a fairly high confidence level in this change after this latest review.  I'm not worried about reverting the whole PR if needed, but if we'd prefer a more incremental approach I think we could do the 2-7 upgrade in one go, then let 7 and 8 coexist for a while before finally removing 7.  Maybe we do that if we run into another issue with it today?

**Pull request merged**